packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -49,6 +57,23 @@ class StorageReference {
   }
 }
 
+class StorageDeleteTask {
+  final String path;
+
+  StorageDeleteTask._(this.path);
+
+  Completer<DeleteTaskResult> _completer = new Completer<DeleteTaskResult>();
+
+  Future<DeleteTaskResult> get future => _completer.future;
+
+  Future<Null> _start() async {
+    final bool result = await FirebaseStorage._channel.invokeMethod(
```

Comment:
I think this will only work when the invocation is successful. When it fails you should catch a PlatformException here.